tests/lib/rules/require-default-props.js

```diff
@@ -1718,6 +1728,17 @@ ruleTester.run('require-default-props', rule, {
           column: 3
         }
       ]
+    },
+    {
+      code: [
+        'import type { ImportedType } from "fake";',
+        'type Props = ImportedType;',
+        'function Hello(props: Props) {',
+        '  return <div>Hello {props.name.firstname}</div>;',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint',
+      errors: []
```

Comment:
Yes, this should be covered by no-undef. The require-default-props rule should just try its bests to resolve the props and the defaultProps, and give up if it cannot do it.

If this test case do not return any error then it should be in the "valid" array.